src/types/list.rs

```diff
@@ -142,16 +153,19 @@ impl PyList {
 /// Used by `PyList::iter()`.
 pub struct PyListIterator<'a> {
     list: &'a PyList,
-    index: isize,
+    index: usize,
```

Comment:
I think with this change there's some conversions in `size_hint` (`self.index as usize`) which aren't needed any more.